src/Services/auth.service.ts

```diff
@@ -0,0 +1,61 @@
+import bcrypt from 'bcrypt';
+
+import bcryptEncrypt from '../Utils/bcrypt.encrypt';
+
+import IUser from '../Interfaces/user.type';
+import AuthRepository from '../Repositories/auth.repository';
+import {
+    BadRequestError, 
+    NotFoundError, 
+    UnAuthenticatedError
+} from "../Utils/ErrorUtils";
+
+export default class AuthService {
+    private authRepository= new AuthRepository()
+
+  public async register(email:string, password: string, user_name: string, phone_number:number, country:string): Promise<IUser> {
```

Comment:
The argument of this function is more than 3 which makes it to long, You can do something like:
` public async register(payload:RegisterI)....
`
where 
```
 RegisterI = {
email:string, 
password: string,
 user_name: string,
 phone_number:number, 
country:string
}
```